### DISCUSSION TITLE
How to run multiple servers with Pocketbase

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3628] [Q&A] [answered]
Short Description:

My website https://stocknear.com/ uses pocketbase for user authentication and also storing watchlist, portfolios, memes etc.
Currently I have a vps server in germany where pocketbase runs. Obviously my user in germany have zero complains about the speed.

However, users in south-america and asia complaint about the lags.

Question:
How can i deploy pocketbase on different servers but still update all pocketbase servers simultaneously.

Any help is much appreciated

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase can't scale horizontally and can work only on a single server.

If you want multiple independent and replicated PocketBase servers (aka. support for both read/write operations), you can try [Marmot](https://github.com/maxpert/marmot) (_I haven't tested it personally; here is a discussion with demo from the author - https://github.com/pocketbase/pocketbase/discussions/487_), **but keep in mind that the app hooks and realtime events will not be synchronized between the instances.** Marmot and other similar replication solutions takes care for "syncing" only the databases.

In general what you are trying to do (_aka. deploy your backend "on the edge"_) is not an easy task and there are always some tradeoffs when building such applications (especially when there is a local state involved) and for most of the time I'd argue that you don't need the complexity that comes with it.

My recommendation is to manually check with a VPN from different locations and monitor the browser's Network tab to ensure that the reported "lags" are caused by requests to the PocketBase server and not by something else (eg. downloading large external assets or API calls).

**Reply by MuslemRahimi** [reply]
Thank you for your response. I will keep it in my mind
